R/foreign-keys.R

```diff
@@ -84,6 +84,21 @@ dm_add_fk <- function(dm, table, columns, ref_table, ref_columns = NULL, ...,
     ref_col_name <- get_key_cols(ref_key)
   } else {
     ref_col_name <- names(eval_select_indices(ref_col_expr, colnames(ref_table_obj)))
+    # check if either a PK or UK already matches ref_col_name
+    all_keys <- dm_get_all_pks_impl(dm) %>%
+      rename(uk_col = pk_col) %>%
+      bind_rows(dm_get_all_uks_impl(dm)) %>%
+      filter(table == ref_table_name)
+    # setequal() could also be used for matching, but IMHO the order should matter
+    matches_keys <- map_lgl(all_keys$uk_col, identical, ref_col_name)
+    if (!any(matches_keys)) {
+      if (check) {
+        if (!is_unique_key_se(ref_table_obj, ref_col_name)$unique) {
+          abort_not_unique_key(ref_table_name, ref_col_name)
+        }
+      }
+      dm <- dm_add_uk_impl(dm, ref_table_name, ref_col_name)
```

Comment:
Do we want to print a message here?

Reply:
I had that originally, but -- mainly due to dm_for_filter() having a UK created each time it's called -- it screwed up our snapshots. Thinking of it, maybe we should repair `dm_for_filter()`, since I don't think that is intentional.
Will add a cli::cli_alert_info() call.

Reply:
Will also do this in a future PR; it seems though, that "repairing" dm_for_filter() causes bigger problems with the snapshots than leaving it the way it is and accepting the UK-message in the affected snapshots.